Userland/Libraries/LibIMAP/QuotedPrintable.cpp

```diff
@@ -52,34 +66,32 @@ ErrorOr<ByteBuffer> decode_quoted_printable(StringView input)
                     u8 actual_character = (parse_ascii_hex_digit(first_escape_character) << 4) | parse_ascii_hex_digit(second_escape_character);
                     TRY(output.try_append(actual_character));
                 } else {
-                    TODO();
+                    TRY(append_invalid_escape_sequence(first_escape_character, second_escape_character));
+                    continue;
                 }
             } else if (first_escape_character == '\r') {
                 if (lexer.is_eof()) {
-                    TODO();
+                    TRY(append_invalid_escape_sequence(first_escape_character));
+                    continue;
                 }
 
                 char second_escape_character = lexer.consume();
 
                 if (second_escape_character == '\n') {
                     // This is a soft line break. Don't append anything to the output.
                 } else {
-                    TODO();
+                    TRY(append_invalid_escape_sequence(first_escape_character, second_escape_character));
+                    continue;
                 }
             } else {
-                if (is_illegal_character(first_escape_character)) {
-                    TODO();
-                }
-
-                // Invalid escape sequence. RFC 2045 says a reasonable solution is just to append '=' followed by the character.
-                TRY(output.try_append('='));
-                TRY(output.try_append(first_escape_character));
+                TRY(append_invalid_escape_sequence(first_escape_character));
+                continue;
             }
         } else {
             TRY(output.try_append(potential_character));
         }
     }
-
+    dbgln("{}", output.string_view());
```

Comment:
Is this line intentionally here? :)

Reply:
No, it was left over from testing :facepalm:. I've removed it.